# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsFrameList.h
# Commit: dd14514b9e99
# Full Hash: dd14514b9e992d986c6802c434e59d67a172f828
# Author: Ting-Yu Lin <tlin@mozilla.com>
# Date: 2022-10-12 09:45:09
# Regressor Bug: 1794456
# File Overlap Count: 1
# Description:
#   Bug 1794456 Part 3 - Support range-based for loop for nsFrameList::Slice. r=emilio
#   
#   Since Slice::mEnd is the first frame that is NOT in the slice, so we only
#   support forward iteration, and it's sufficient to replace existing usages of
#   nsFrameList::Enumerator.
# ==============================================================================

diff -r 58733fee05e0 -r dd14514b9e99 layout/generic/nsFrameList.h
--- a/layout/generic/nsFrameList.h	Tue Oct 11 21:01:27 2022 +0000
+++ b/layout/generic/nsFrameList.h	Tue Oct 11 21:01:28 2022 +0000
@@ -377,10 +377,17 @@
         : mStart(aList.FirstChild()), mEnd(nullptr) {}
     Slice(nsIFrame* aStart, nsIFrame* aEnd) : mStart(aStart), mEnd(aEnd) {}
 
+    iterator begin() const { return iterator(mStart); }
+    const_iterator cbegin() const { return begin(); }
+    iterator end() const { return iterator(mEnd); }
+    const_iterator cend() const { return end(); }
+
    private:
-    nsIFrame* const mStart;      // our starting frame
-    const nsIFrame* const mEnd;  // The first frame that is NOT in the slice.
-                                 // May be null.
+    // Our starting frame.
+    nsIFrame* const mStart;
+
+    // The first frame that is NOT in the slice. May be null.
+    nsIFrame* const mEnd;
   };
 
   class Enumerator {